### PR TITLE
Sell for collateral now supports slippage

### LINKED ISSUE #451: Update sellForCollateral to use an array or swaps in vs a static sets out value
To support this we need to update sellForCollateral to use an array or swaps in vs a static sets out value.

### PR DESCRIPTION
**Pull request by justinbarry**
close #451

### CODE REVIEW

File: packages/smart/test/amm-factory-test.ts [hunk 1]
```diff
@@ -125,8 +125,8 @@ describe("AMMFactory", () => {
       secondMarketFactory.address,
       marketId,
       _outcome,
-      _setsInForCollateral,
-      setsToBurn
+      _shareTokensIn.map((m) => BigNumber.from(m)),
+      BigNumber.from(0)
```
Comment (robert-e-davidson3) [inline packages/smart/test/amm-factory-test.ts]: I'm worried about this being zero. The test of the shareTokensIn calculation is how close it gets to minSetsOut. So the minSetsOut value needs to match what we want the UI to calculate when it figures out slippage tolerance.

**Review by robert-e-davidson3 — CHANGES_REQUESTED** [review]
inline

File: packages/smart/test/amm-factory-test.ts (hunk 1 above)
Comment (justinbarry) [reply packages/smart/test/amm-factory-test.ts]: Good call.

File: packages/smart/test/amm-factory-test.ts [hunk 2]
```diff
@@ -125,12 +125,14 @@ describe("AMMFactory", () => {
...
       _outcome,
-      _setsInForCollateral,
-      setsToBurn
+      _shareTokensIn.map((m) => BigNumber.from(m)),
+      BigNumber.from(tokenAmountOut)
     );
 
     const collateralAfter = await secondCollateral.balanceOf(secondSigner.address);
 
+    console.log("collateralAfter", collateralAfter.toString());
```
Comment (robert-e-davidson3) [inline packages/smart/test/amm-factory-test.ts]: debug code?

**Review by robert-e-davidson3 — APPROVED** [review]

**Pull request merged**